src/main/java/igym/entities/User.java

```diff
@@ -7,30 +7,38 @@
 @Table(name = "users")
 @Entity
 public class User {
-    
+
     @Id
     @GeneratedValue(strategy = GenerationType.UUID)
     private UUID id;
     private String name;
 
-    public User(UUID id, String name) {
-        this.id = id;
+    public User() {
+    }
+
+    public User(String name) {
         this.name = name;
     }
 
     public UUID getId() {
         return id;
     }
 
-    public void setId(UUID id) {
-        this.id = id;
-    }
-
     public String getName() {
         return name;
     }
 
     public void setName(String name) {
         this.name = name;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null || getClass() != obj.getClass())
+            return false;
+        User other = (User) obj;
+        return id != null && id.equals(other.id);
+    }
 }
```

Comment:
maybe this method is useful for another route if we are not using better to take it out